### PR TITLE
28 similar student

### PR DESCRIPTION
**Pull request by acmccuaig**
Working similar student function and API. Takes in a student and section and returns a list of 5 similar students.

### CODE REVIEW

**Comment by andrew-wagner89** [comment]
relies on #37, use this branch for testing 18-drag-and-drop on the front end

File: src/api/index.js [hunk 1]
```diff
@@ -4,11 +4,13 @@ import placements from './placements'
 import sections from './sections'
 import grades from './grades'
 import students from './students'
+import similarStudent from './similar-student'
 
 export default ({ config, db }) => {
 	let api = Router()
 
 	api.use('/placements', placements({ config, db }))
+	api.use('/similar-student', similarStudent({ config, db }))
```
Comment (jamesvclements) [inline src/api/index.js]: Just a thought - `api.use('students/similar', similarStudent({ config, db} ))` might be a little nicer... e.g. having the similar functionality within the `students` endpoint. IF it works

File: src/services/similar-student/similar-student.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	
+	for ( let i = 0; i < swapSection.students.length; i++ ){
+		swapSection.students[i].dif =  Math.abs(swapStudent.weightedScore - swapSection.students[i].weightedScore)
+		if(swapStudent.sex === swapSection.students[i].sex)
+			similarStudents.push(swapSection.students[i])
+	}
+	
+	similarStudents.sort((a, b) => { return b.dif - a.dif})
+	
+	similarStudents = similarStudents.slice(0,5)
```
Comment (jamesvclements) [inline src/services/similar-student/similar-student.js]: for brevity you could just do `return similarStudents.slice(0, 5)`

File: src/api/index.js (hunk 1 above)
Comment (andrew-wagner89) [reply src/api/index.js]: I think it'd be better to stick with /similar-student. We ran into some trouble with resource-router-middleware and having multiple levels of endpoints, and I figured our time was better spent on other problems

**Comment by andrew-wagner89** [comment]
looks good to me

**Pull request merged**